### ISSUE TITLE
pydantic2-resolve doesn't seem to support more specific types

### ISSUE THREAD

**Issue by SuperDoxin** [allmonday/pydantic2-resolve#7]
It seems that types like `tuple[int,int]` do not work, while `tuple` works perfectly fine. Somewhat oddly enough `tuple | None` does not work, while `Optional[tuple]` *does*. Reproducible using this code:

```
import asyncio
from pydantic import BaseModel, Field
from pydantic_resolve import Resolver

class Test(BaseModel):
    value: tuple[int, int]=(0,0)

async def test():
    return await Resolver().resolve(Test())

async def main():
    print(await test())

asyncio.run(main())
```

which produces this traceback:
```
Traceback (most recent call last):
  File "/home/my-user/pydantic-test/src/example.py", line 14, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/my-user/pydantic-test/src/example.py", line 12, in main
    print(await test())
          ^^^^^^^^^^^^
  File "/home/my-user/pydantic-test/src/example.py", line 9, in test
    return await Resolver().resolve(Test())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/my-user/pydantic-test/venv/lib/python3.11/site-packages/pydantic_resolve/resolver.py", line 288, in resolve
    self.scan_data = core.scan_and_store_required_fields(target)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/my-user/pydantic-test/venv/lib/python3.11/site-packages/pydantic_resolve/core.py", line 31, in scan_and_store_required_fields
    update_forward_refs(root)
  File "/home/my-user/pydantic-test/venv/lib/python3.11/site-packages/pydantic_resolve/util.py", line 274, in update_forward_refs
    update_pydantic_forward_refs(kls)
  File "/home/my-user/pydantic-test/venv/lib/python3.11/site-packages/pydantic_resolve/util.py", line 261, in update_pydantic_forward_refs
    update_forward_refs(shelled_type)
  File "/home/my-user/pydantic-test/venv/lib/python3.11/site-packages/pydantic_resolve/util.py", line 273, in update_forward_refs
    if issubclass(kls, BaseModel):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen abc>", line 123, in __subclasscheck__
TypeError: issubclass() arg 1 must be a class
```

The issue seems to be that issubclass is getting called with the type as argument, which doesn't work if the type is not a class. I'm not entirely sure what the right fix here might be. Catching the error and simply assuming issubclass would return False if the type isn't a class may work, but I'm not familiar enough with the code to be sure.

**Comment by allmonday** [comment]
confirmed.

thank you for your issue.

inside pydantic-resolve, it has a `shelling_type` func to get the inner class from list or optional.

union is not supported yet, so that Optional[tuple] works but tuple | None failed. (union a None is a special form of Optional)

issubclass raises TypeError when param is not class, I'll fix this in next release.

**Comment by SuperDoxin** [comment]
Awesome, thanks for the quick response! Let me know if there's anything I can help with.

**Comment by allmonday** [comment]
@SuperDoxin 
issue of issubclass is fixed, in v2.1.0
